### PR TITLE
add sqlcipher/4.4.0

### PR DESCRIPTION
**Pull request by theodelrieu**
Specify library name and version:  **sqlcipher/4.4.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
Hi @theodelrieu could you explain a little bit about those patches? Are they from the upstream?

**Comment by theodelrieu** [comment]
Long story short: The windows (nmake) build is broken, but most of the change are about library names, which can be removed (even though I find confusing to have `sqlite3.lib`)

I don't remember why all those patches were needed to be honest, I had some cryptic build errors a few years ago with 3.4.2, and spent a few days making it work. It included bad calling convention (had to patch it to `cdecl`), linking the debug system libs in release and vice versa etc.

Once I'll make it work on my personal CI, I'll try to remove as many of those patches as I can though.

**Comment by theodelrieu** [comment]
@uilianries I've rewrote the patch, adding explanation here https://github.com/conan-io/conan-center-index/commit/c7a91f473d02c6fcd10527889cae4063329ce660

**Comment by theodelrieu** [comment]
Note that using the CCI `tcl/8.6.10` as a build requirement will fail when cross-building.

This is because of the following lines in `tcl`:

```python
        if self.settings.os not in ("Linux", "Macos", "Windows"):
            raise ConanInvalidConfiguration("Unsupported os")
```

**Comment by theodelrieu** [comment]
Here you go! Let me know if you have some questions about the patches

**Comment by theodelrieu** [comment]
Funny thing, there is a configure flag `--disable-largefile`, but it doesn't set `SQLITE_DISABLE_LFS`, and is therefore useless

File: recipes/sqlcipher/all/conanfile.py [hunk 1]
```diff
@@ -74,13 +108,15 @@ def _build_autotools(self):
         autotools_env = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
         if self.settings.os == "Linux":
             autotools_env.libs.append("dl")
+            if not self.options.with_largefile:
```
Comment (uilianries) [inline recipes/sqlcipher/all/conanfile.py:111]:
```suggestion
            if not self.options.get_safe("with_largefile"):
```

Comment (theodelrieu) [reply recipes/sqlcipher/all/conanfile.py:111]: I don't think this is needed, since the option is deleted when `settings.os != "Linux"`

**Pull request merged**